File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let square n = pown n 2

let squareOfSum (number: int): int = seq { 1 .. number } |> Seq.sum |> square

let sumOfSquares (number: int): int = seq { 1 .. number } |> Seq.sumBy square

let differenceOfSquares (number: int): int = (squareOfSum number) - (sumOfSquares number)

File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number)
{
    unsigned int sum = 0;
    for (unsigned int n = 1; n <= number; n++)
    {
        sum += n * n;
    }
    return sum;
}

unsigned int square_of_sum(unsigned int number)
{
    unsigned int sum = 0;
    for (unsigned int n = 1; n <= number; n++)
    {
        sum += n;
    }
    return sum * sum;
}

unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}
